### PR TITLE
Release updates

### PR DESCRIPTION
**Pull request by lrknox**
Add release date and known problems entry for no Fortran in Windows binaries.

### CODE REVIEW

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -615,6 +615,9 @@ Platforms Tested
 Known Problems
 ==============
 
+    Building HDF5 Fortran on windows with Intel oneAPI 2023.2 currently fails 
+    with link errors.  Windows binaries will be without Fortran for this release.
+
```
Comment (brtnfld) [inline release_docs/RELEASE.txt]: Building HDF5 Fortran on Windows with Intel oneAPI 2023.2 currently fails for this release with link errors. As a result, this release of Windows binaries will not include Fortran.

Comment (lrknox) [inline release_docs/RELEASE.txt:620]:
```suggestion
 Building HDF5 Fortran on Windows with Intel oneAPI 2023.2 currently fails for this release with link errors. As a result, this release of Windows binaries will not include Fortran.
```

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -615,6 +615,9 @@ Platforms Tested
 Known Problems
 ==============
 
+    Building HDF5 Fortran on windows with Intel oneAPI 2023.2 currently fails 
+    with link errors.  Windows binaries will be without Fortran for this release.
```
Comment (lrknox) [inline release_docs/RELEASE.txt:619]:
```suggestion
Building HDF5 Fortran on Windows with Intel oneAPI 2023.2 currently fails for this release with link errors. As a result, this release of Windows binaries will not include Fortran.
```

Comment (lrknox) [inline release_docs/RELEASE.txt:619]: Building HDF5 Fortran on Windows with Intel oneAPI 2023.2 currently fails for
    this release with link errors. As a result, this release of Windows binaries will not
    include Fortran.

Comment (lrknox) [inline release_docs/RELEASE.txt:619]:
```suggestion
    Building HDF5 Fortran on windows with Intel oneAPI 2023.2 currently fails 
    with link errors.  Windows binaries will be without Fortran for this release.
```

**Review by brtnfld — APPROVED** [review]

**Pull request merged**